aptos-move/framework/aptos-framework/sources/configs/features.move

```diff
@@ -0,0 +1,126 @@
+/// Maintains feature flags.
+module aptos_framework::features {
+    use aptos_framework::system_addresses;
+    use std::vector;
+
+    // ============================================================================================
+    // Feature Flag Definitions
+
+    // Each feature flag should come with documentation which justifies the need of the flag.
+    // Introduction of a new feature flag requires approval of framework owners. Be frugal when
+    // introducing new features, as too many features can make it hard to understand the code.
+    //
+    // Each feature flag should come with a specification of a lifetime:
+    //
+    // - a *transient* feature flag is only needed until a related code rollout has happened. This
+    //   is typically associated with the introduction of new native Move functions, and is only used
+    //   from Move code. The owner of this feature is obliged to remove it once this can be done.
+    //
+    // - an *ephemeral* feature flag is required to stay around forever. Typically, those flags guard
+    //   behavior in native code, and the behavior with or without the feature need to be preserved
+    //   for playback.
+    //
+    // Features should be grouped along components. Their name should start with the module or
+    // component name they are associated with.
+
+
+    // --------------------------------------------------------------------------------------------
+    // Code Publishing
+
+    /// Whether validation of package dependencies is enabled, and the related native function is
+    /// available. This is needed because of introduction of a new native function.
+    /// Lifetime: transient
+    const CODE_DEPENDENCY_CHECK: u64 = 1;
+    public fun code_dependency_check_enabled(): bool acquires Features {
+        is_enabled(CODE_DEPENDENCY_CHECK)
+    }
+
+    // ============================================================================================
+    // Feature Flag  Management
+
+    /// The enabled features, represented by a bitset stored on chain.
+    struct Features has key {
+        features: vector<u8>,
```

Comment:
Why not use a table? vector would be loaded entirely so adding more features would increase the cost for all existing feature checks

Reply:
This seems way to inefficient. Notice if we have some 256 feature flags, right now we only need 8 bytes to represent them.